client/src/Login.js

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react';
+import { withRouter } from 'react-router-dom';
+
+import axios from './helpers/axiosConfig';
+
+class Login extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      username: '',
+      password: '',
+    }
+
+    this.defaultState = this.state;
+
+  }
+
+  handleChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value,
+
+    })
+  }
+
+  handleSubmit = async (event) => {
+    event.preventDefault();
+    const {username, password} = this.state;
+
+    try {
+
+      const result = await axios.post('/login', {
+        username,
+        password,
+      });
+
+      const token = result.data.authToken;
+      localStorage.setItem('token', token);
+
+      this.props.history.push('/jokes');
+
+    }
+    catch (err) {
+      console.error(err);
+
+    }
+
+    this.setState({
+      ...this.defaultState,
+
+    })
+  }
+
+  render() {
+
+    return (
+      <>
```

Comment:
some time using fractures is unavoidable but often I would suggest looking into methods to avoid this as it can present we're styling issues that can be difficult to debug. Try going with just a section or something semantic that isn't naturally display block.